File: Modul 6/soal 4/PRAK604-2410817210011-MuhammadDzulFathiAhyan.py
hestek=0 
bitang=0 
kawannya_naruto =input() 
musuhnya=input() 
if len(kawannya_naruto) != len(musuhnya): 
    print("Panjang kalimat berbeda, pesan palsu")
    exit()
for i in range(len(kawannya_naruto)): 
    if kawannya_naruto[i] == ' ' and musuhnya[i] == ' ': 
        continue 
    if musuhnya[i] == kawannya_naruto[i]: 
        print("*", end="") 
        bitang += 1 
    else: 
        print("#", end="") 
        hestek += 1 
print("\n* = ", bitang) 
print("# = ",  hestek) 
if (bitang >= hestek): print("Pesan Asli") 
else: print("Pesan Palsu")

File: Modul 6/soal 4/PRAK604-2410817210011-MuhammadDzulFathiAhyan.c
#include <stdio.h> 
int panjang_kalimat(char kalimat[])
{ 
    int panjang = 0; 
    while(kalimat[panjang] != '\0') panjang++; 
    return panjang; 
}
int main()
{ 
    int bintang = 0, hestek = 0; 
    char shikamaru[99]; 
    char musuhnya[99]; 
    scanf(" %[^\n]", shikamaru); 
    scanf(" %[^\n]", musuhnya); 
    if(panjang_kalimat(shikamaru) != panjang_kalimat(musuhnya))
    { 
        printf("Panjang kalimat berbeda, pesan palsu"); 
        return 1; 
    } 
    for(int i = 0; i < panjang_kalimat(shikamaru); i++)
    { 
        if(shikamaru[i] == ' ' && musuhnya[i] == ' '){continue;} 
 
        if(shikamaru[i] == musuhnya[i])
        { 
            printf("*"); 
            bintang++; 
        } 
        else
        { 
            printf("#"); 
            hestek++; 
        } 
    } 
    printf("\n* = %d \n", bintang); 
    printf("# = %d \n", hestek); 
    if(bintang >= hestek) printf("Pesan Asli"); 
    else printf("Pesan Palsu"); 
    return 0; 
} 